### PR TITLE
CodeGuru engine updates

### PR DESCRIPTION
**Pull request by Yovaayova**
here is the engine new features, in a more logical way. I hope it works...
The new changes are starting after the commit described as "In order to make the changes checking easier."

### CODE REVIEW

**Comment by rmkn85** [comment]
This PR is quite old, why wasn't it merged? admins?

**Comment by rmkn85** [comment]
@Yovaayova hi, i want to merge some of your commits, are you still active here?

**Comment by Yovaayova** [comment]
Yeh, sure.

File: corewars8086/.classpath [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.7"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.6"/>
```
Comment (rmkn85) [inline corewars8086/.classpath]: why downgrade from java 7 to java 6?

**Comment by rmkn85** [comment]
Reviewed, changes look good.
To sum up, you:
1. adjusted window placement
2. fixed byte/word overflow bug in shift operations
3. enabled running subset of warriors and in random order
Is that correct?

Why downgrade to java 6? If it is not needed, please revert that part, and then I'll merge.

**Pull request merged**